controllers/managedocs_controller_test.go

```diff
@@ -61,6 +72,86 @@ var _ = Describe("ManagedOCS controller", func() {
 				Expect(k8sClient.Create(ctx, managedOCS)).Should(Succeed())
 				Expect(k8sClient.Get(ctx, getResourceKey(managedOCS), managedOCS)).Should(Succeed())
 
+				// No storage cluster should be created
+				scList = &ocsv1.StorageClusterList{}
+
+				Expect(k8sClient.List(ctx, scList, client.InNamespace(TestNamespace))).Should(Succeed())
+				Expect(scList.Items).Should(HaveLen(0))
```

Comment:
Testing the existence of a storage cluster immediately after the creation of the ManagedOCS is problematic as you don't know if the storage cluster was not created because of the lack of the secret or just because the reconcile loop did not end yet. 

My suggestion is to wait until the ManagedOCS has a status section to insure full reconciliation then validate that there is no storageclsuter.

On the positive cases I am using `waitForReasource` to a accomplish this

Reply:
While a valid point , I believe this can be done in a separate PR.